File: model/src/main/java/com/github/fburato/highwheelmodules/model/classpath/ClasspathRoot.java
package com.github.fburato.highwheelmodules.model.classpath;

import java.io.IOException;
import java.io.InputStream;
import java.util.Collection;

import com.github.fburato.highwheelmodules.model.bytecode.ElementName;

public interface ClasspathRoot {

    InputStream getData(ElementName name) throws IOException;

    Collection<ElementName> classNames();

    InputStream getResource(final String name) throws IOException;
}

File: model/src/main/scala/com/github/fburato/highwheelmodules/model/classpath/ClasspathRoot.scala
package com.github.fburato.highwheelmodules.model.classpath

import com.github.fburato.highwheelmodules.model.bytecode.ElementName

import java.io.InputStream
import scala.util.Try


trait ClasspathRoot {

  def getData(elementName: ElementName): Try[InputStream]

  def classNames: Try[Seq[ElementName]]

  def getResource(name: String): Try[InputStream]
}
